src/main/java/org/wso2/carbon/module/swiftiso20022/ISO20022Camt053Validator.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.module.swiftiso20022;
+
+import org.apache.synapse.MessageContext;
+import org.jaxen.JaxenException;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+import org.wso2.carbon.module.swiftiso20022.utils.ConnectorUtils;
+import org.wso2.carbon.module.swiftiso20022.utils.ISO20022Camt053ValidatorUtils;
+import org.wso2.carbon.module.swiftiso20022.utils.ISOMessageParser;
+import org.wso2.carbon.module.swiftiso20022.utils.XSDValidator;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+
+import java.io.IOException;
+
+/**
+ * Validate the ISO20022.camt.053 message.
+ */
+public class ISO20022Camt053Validator extends AbstractConnector {
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        boolean isBusinessMsg = false;
+        String rootElementTag = ISOMessageParser.getRootXMLElement(messageContext);
+
+        if (!ConnectorConstants.XML_INPUT_BUSINESS_ENV_TAG.equals(rootElementTag) &&
+                (!ConnectorConstants.XML_INPUT_DOCUMENT_TAG.equals(rootElementTag))) {
+            // Invalid XML root tag
+            this.log.error(ConnectorConstants.ERROR_INVALID_XML_ROOT_TAG);
```

Comment:
Have we checked,if this is the recommended way to add logs 'this.log'. I noticed there's a class called SynapseLog and some of the existing connectors refer it.If possible,check with a MI team member.

Reply:
Fixed in 7fad8deab23338ef750aaf8bf737ebee9dcb763d